viewing_party/party.py

```diff
@@ -1,23 +1,163 @@
 # ------------- WAVE 1 --------------------
 
+# Let's start
+# Hola, Laura!
+
 def create_movie(title, genre, rating):
-    pass
+    title_check = isinstance(title, str)
+    genre_check = isinstance(genre, str)
+    rating_check = isinstance(rating, float)
+    if not title_check or not genre_check or not rating_check:
+        return None
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    return movie
+    
+
+def add_to_watched(user_data, movie):
+    user_data_watched_list = user_data["watched"]
+    user_data_watched_list.append(movie)
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data_watchlist = user_data["watchlist"]
+    user_data_watchlist.append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    lengh_list = len(user_data["watched"])
+    if lengh_list < 1:
+        return 0.0
+    rating_added = 0
+    for element in user_data["watched"]:
+        rating_added += element["rating"]
+    average_rating = rating_added/lengh_list
+    return float(average_rating)
+
 
+def get_most_watched_genre(user_data):
+    count_watched_genres = {}
+    if user_data["watched"] == []:
+        return None
+    for element in user_data["watched"]:
+        if element["genre"] not in count_watched_genres:
+            count_watched_genres[element["genre"]] = 1

```

Comment:
This should be set to 0 since you will add 1 to it on the next line. The way this code works, all genres start out with '2' the first time you see it. Since all genres get this, it doesn't impact the functionality of the code, but it is confusing.